base/src/main/java/io/spine/validate/NumberText.java

```diff
@@ -80,35 +78,37 @@
      * @return whether this instance of a number is of the same {@code Number} subtype as
      *         the specified one.
      */
-    boolean isOfSameType(StringDescribedNumber anotherNumber) {
-        Class<? extends Number> classOfThisNumber = numericValue.getClass();
-        Class<? extends Number> classOfAnotherNumber = anotherNumber.numericValue.getClass();
+    boolean isOfSameType(NumberText anotherNumber) {
+        Class<? extends Number> classOfThisNumber = value.getClass();
+        Class<? extends Number> classOfAnotherNumber = anotherNumber.value.getClass();
         return classOfThisNumber.equals(classOfAnotherNumber);
     }
 
-    private static Number parseNumber(String stringRepresentation) {
+    private static Number parseNumber(String text) {
         ImmutableList<String> wholeAndDecimal =
-                ImmutableList.copyOf(DECIMAL_SPLIT.split(stringRepresentation));
+                ImmutableList.copyOf(DECIMAL_SPLIT.split(text));
         hasOnlyWholeAndDecimal(wholeAndDecimal);
-        if (hasDecimalPart(stringRepresentation)) {
-            return Double.parseDouble(stringRepresentation);
+        if (hasDecimalPart(text)) {
+            return Double.parseDouble(text);
+        }
+        if (fitsIntoInteger(text)) {
```

Comment:
The `if` branch is only partially tested. It would be tested in full once you test the negative outcome.